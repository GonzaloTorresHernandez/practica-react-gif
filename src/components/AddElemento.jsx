import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddElemento = ({ onNewValor }) => {
  const [valorInput, setValorInput] = useState("");

  const onInputChange = (event) => {
    setValorInput(event.target.value);
  };

  const enviarForm = (event) => {
    event.preventDefault();
    if (valorInput.trim().length <= 1) return;
    onNewValor(valorInput.trim());
    setValorInput("");
  };

  return (
    <>
      <form onSubmit={enviarForm} aria-label="form">
        <input
          type="text"
          id="textoIngresado"
          placeholder="Ingrese el anime..."
          value={valorInput}
          onChange={(event) => onInputChange(event)}
        />
      </form>
    </>
  );
};

AddElemento.propTypes = {
    onNewValor: PropTypes.func.isRequired,
}
