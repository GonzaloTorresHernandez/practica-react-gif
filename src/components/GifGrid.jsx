import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ valor }) => {

    const { imagenes, estaCargando} = useFetchGifs(valor);

    return (
        <>
            <h3>{valor}</h3>
            {
                // llamar solo si se cumple el estaCargando
                estaCargando && (<h4>Cargando...</h4>)
            }
            
            <div className="card-grid">
                {
                    imagenes.map(({ id, titulo, url }) => {
                        return (
                            <GifItem key={id} titulo={titulo} url={url} />
                        );
                    })
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    valor: PropTypes.string.isRequired
}