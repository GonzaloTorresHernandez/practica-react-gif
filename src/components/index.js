/*
    archivo de barril
    para conteneder todos las importacion de la carpeta
*/

export * from './AddElemento';
export * from './GifGrid';
export * from './GifItem';