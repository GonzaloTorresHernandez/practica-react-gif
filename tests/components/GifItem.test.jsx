import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {

    const titulo = 'titulo';
    const url = 'https://mi-url.com/img.jpg';

    test('prueba martch con el stanpshot', () => {
        const { container } = render(
            <GifItem titulo={ titulo } url={ url } />
        );
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem titulo={ titulo } url={ url } />);

        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( titulo );

        const { src, alt } = screen.getByRole('img');
        expect(alt).toBe(titulo);
        expect(src).toBe(url);
    });

    test('Debe de mostar el titulo en el componente', () => {
        render(<GifItem titulo={ titulo } url={ url } />);
        expect( screen.getByAltText(titulo)).toBeTruthy();
    });
});