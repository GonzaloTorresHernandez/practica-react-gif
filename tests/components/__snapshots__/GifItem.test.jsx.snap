// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem prueba martch con el stanpshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://mi-url.com/img.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
